debug/TorrentClient/message.py

```diff
@@ -0,0 +1,324 @@
+
+    def __eq__(self, other):
+        return self.to_bytes() == other.to_bytes()
+
+
+class KeepAlive(Message):
+    '''keep-alive: <len=0000>
+
+    The keep-alive message is a message with zero bytes, specified with
+    the length prefix set to zero. There is no message ID and no
+    payload. Peers may close a connection if they receive no messages
+    (keep-alive or any other message) for a certain period of time, so a
+    keep-alive message must be sent to maintain the connection alive if
+    no command have been sent for a given amount of time. This amount of
+    time is generally two minutes.
+    '''
+    def __init__(self):
+        Message.__init__(self)
+        self.length = 0
+
+
+class Choke(Message):
+    '''choke: <len=0001><id=0>
+
+    The choke message is fixed-length and has no payload.
+    '''
+    def __init__(self):
+        Message.__init__(self)
+        self.length = 1
+        self.id = 0
+
+
+class UnChoke(Message):
+    '''unchoke: <len=0001><id=1>
+
+    The unchoke message is fixed-length and has no payload.
+    '''
+    def __init__(self):
+        Message.__init__(self)
+        self.length = 1
+        self.id = 1
+
+
+class Interested(Message):
+    '''interested: <len=0001><id=2>
+
+    The interested message is fixed-length and has no payload.
+    '''
+    def __init__(self):
+        Message.__init__(self)
+        self.length = 1
+        self.id = 2
+
+
+class NotInterested(Message):
+    '''not interested: <len=0001><id=3>
+
+    The not interested message is fixed-length and has no payload.
+    '''
+    def __init__(self):
+        Message.__init__(self)
+        self.length = 1
+        self.id = 3
+
+
+class Have(Message):
+    '''have: <len=0005><id=4><piece index>
+
+    The have message is fixed length. The payload is the zero-based
+    index of a piece that has just been successfully downloaded and
+    verified via the hash.
+    '''
+    def __init__(self, index):
+        self.length = 5
+        self.id = 4
+        self.payload = index
+
+    def to_bytes(self):
+        return super(Have, self).to_bytes() + pack('!l', self.payload)
+
+
+class BitField(Message):
+    def __init__(self, payload):
+        raise Exception('Not Implemented')
+
+
+class Request(Message):
+    '''request: <len=0013><id=6><index><begin><length>
+
+    The request message is fixed length, and is used to request a
+    block. The payload contains the following information:
+
+    index: integer specifying the zero-based piece index
+    begin: integer specifying the zero-based byte offset within the
+    piece
+    length: integer specifying the requested length.
+    '''
+    def __init__(self, index, begin, length):
+        Message.__init__(self)
+        self.length = 13
+        self.id = 6
+        self.index = index
+        self.begin = begin
+        self.request_length = length
+
+    def to_bytes(self):
+        r = super(Request, self).to_bytes()
+        r += pack('!l', self.index)
```

Comment:
From https://docs.python.org/3/library/struct.html:

The form '!' is available for those poor souls who claim they can’t remember whether network byte order is big-endian or little-endian.

Hah :P